fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -418,8 +419,315 @@ public void testIndexFromIndexPersistentPreventBadContinuation() {
     }
 
     @Test
-    public void testIndexFromIndexRebuild() {
+    public void testIndexFromIndexPersistentContinuePreviousByIndex() {
+        // start indexing by Index, verify refusal to continue by records, then allow continuation of the previous by index method - overriding the by records request
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final int numRecords = 1328;
+        final int chunkSize  = 42;
+
+        Index srcIndex = new Index("src_index", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index("tgt_index", field("num_value_3_indexed"), IndexTypes.VALUE);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = myHook(srcIndex, tgtIndex);
+
+        openSimpleMetaData();
+        populateData(numRecords);
+
+        openSimpleMetaData(hook);
+        buildSrcIndex(srcIndex);
+
+        openSimpleMetaData(hook);
+        final AtomicLong counter = new AtomicLong(0);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
```

Comment:
It feels like this part of the index build test code should be extractable into a method. Maybe `crashDuringBuild` or something that takes an `IndexFromIndexPolicy` and then creates an `OnlineIndexer` that then uses the policy and crashes halfway through?

May decrease the code dupe in these tests, which is a little high atm.

Reply:
Used `buildIndexAndCrashHalfway`.